src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -808,6 +814,14 @@ private boolean handleJmxNotificationForCassandra22(
                     .withEndTime(DateTime.now())
                     .withId(segmentId)
                     .build());
+
```

Comment:
can we add an `assert` here that failed hasn't already been marked true.

Reply:
again